Octokit/Http/JsonHttpPipeline.cs

```diff
@@ -44,11 +46,18 @@ public void DeserializeResponse<T>(IResponse<T> response)
 
             if (response.ContentType != null && response.ContentType.Equals("application/json", StringComparison.Ordinal))
             {
+                var body = response.Body;
                 // simple json does not support the root node being empty. Will submit a pr but in the mean time....
-                if (response.Body != "{}") 
+                if (body != "{}") 
                 {
-                    var json = _serializer.Deserialize<T>(response.Body);
-                    response.BodyAsObject = json;
+                    // If we're expecting an array, but we get a single object, just wrap it.
+                    // This supports an api that dynamically changes the return type based on the content.
+                    if ((typeof(IEnumerable).IsAssignableFrom(typeof(T)))
+                        && body.StartsWith("{", StringComparison.Ordinal))
```

Comment:
Paranoia check: could this break if we get whitespace around the `{}`? (it's probably unlikely)